components/datetime/src/provider/mod.rs

```diff
@@ -193,5 +200,82 @@ pub mod gregory {
             pub medium: Cow<'static, str>,
             pub short: Cow<'static, str>,
         }
+
+        /// This struct is a public wrapper around the internal Pattern struct. This allows
+        /// access to the serialization and deserialization capabilities, without exposing the
+        /// internals of the pattern machinery.
+        ///
+        /// The Pattern is an "exotic type" in the serialization process, and handles its own
+        /// custom serialization practices.
+        #[derive(Debug, PartialEq, Clone, Default)]
+        #[cfg_attr(
+            feature = "provider_serde",
+            derive(serde::Serialize, serde::Deserialize)
+        )]
+        pub struct PatternV1(pub Pattern);
+
+        impl From<Pattern> for PatternV1 {
+            fn from(pattern: Pattern) -> Self {
+                Self(pattern)
+            }
+        }
+
+        impl TryFrom<&str> for PatternV1 {
+            type Error = pattern::Error;
+
+            fn try_from(pattern_string: &str) -> Result<Self, Self::Error> {
+                let pattern = Pattern::from_bytes(pattern_string);
+                match pattern {
+                    Ok(pattern) => Ok(PatternV1::from(pattern)),
+                    Err(err) => Err(err),
+                }
+            }
+        }
+
+        /// This struct is a public wrapper around the internal Skeleton struct. This allows
+        /// access to the serialization and deserialization capabilities, without exposing the
+        /// internals of the skeleton machinery.
+        ///
+        /// The Skeleton is an "exotic type" in the serialization process, and handles its own
+        /// custom serialization practices.
+        #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+        #[cfg_attr(
+            feature = "provider_serde",
+            derive(serde::Serialize, serde::Deserialize)
+        )]
+        pub struct SkeletonV1(pub Skeleton);
+
+        impl TryFrom<&str> for SkeletonV1 {
+            type Error = SkeletonError;
+
+            fn try_from(skeleton_string: &str) -> Result<Self, Self::Error> {
+                match Skeleton::try_from(skeleton_string) {
+                    Ok(skeleton) => Ok(SkeletonV1(skeleton)),
+                    Err(SkeletonError::FieldsOutOfOrder(fields)) => {
+                        // The fields were out of order, converting fields to a skeleton will ensure
+                        // that these fields are sorted. This sorting is only done in the provider.
+                        Ok(SkeletonV1(Skeleton::from(fields)))
```

Comment:
Observation: Although the sorting is only done in the provider, this impl requires the sorting *code* to be shipped regardless.  If you apply my suggestion above, then you won't need this code any more.